tcptracer-bpf.c

```diff
@@ -854,43 +896,59 @@ static int tcp_send(struct pt_regs *ctx, const size_t size) {
 		return 0;
 	}
 
-	if (status->protocol_inspection_enabled) {
-		struct msghdr msg = {};
-		bpf_probe_read(&msg, sizeof(msg), k_msg);
-		if ((msg.msg_iter.type & ~(READ | WRITE)) == status->iter_type) {
-			char *data = bpf_map_lookup_elem(&write_buffer_heap, &zero);
-			if (data != NULL) {
-				struct iovec iov = {};
-				bpf_probe_read(&iov, sizeof(iov), (void *)msg.msg_iter.iov);
-				bpf_probe_read(data, MAX_MSG_SIZE, iov.iov_base);
-
-				struct ipv4_tuple_t t = {};
-				if (check_family(sk, status, AF_INET)) {
-					if (read_ipv4_tuple(&t, status, sk)) {
-						t.lport = ntohs(t.lport); // Making ports human-readable
-						t.rport = ntohs(t.rport);
-
-						struct tracked_socket *res = bpf_map_lookup_elem(&tracked_sockets, &sk);
-						u64 current_time = bpf_ktime_get_ns();
-						u64 ttfb = 0;
-						struct tracked_socket tsocket = {.active = 1};
-						tsocket.prev_send_time_ns = current_time;
-						bpf_map_update_elem(&tracked_sockets, &sk, &tsocket, BPF_ANY);
-						if (res != NULL) {
-							ttfb = current_time - res->prev_send_time_ns;
-						}
-						u64 cpu = bpf_get_smp_processor_id();
-						int http_status_code = 0;
-						u16 mysql_greeting_protocol_version = 0;
-						if (parse_http_response(data, iov.iov_len, &http_status_code)) {
-							send_http_response(ctx, t, http_status_code, ttfb / 1000, current_time, cpu);
-						} else if (parse_mysql_greeting(data, iov.iov_len, &mysql_greeting_protocol_version)) {
-							send_mysql_greeting(ctx, t, mysql_greeting_protocol_version, current_time, cpu);
-						}
-					}
-				}
-			}
-		}
+    if (status->protocol_inspection_enabled) {
+        struct msghdr msg = {};
+        bpf_probe_read(&msg, sizeof(msg), k_msg);
+    //	bpf_debug("test %d <> %d\n", msg.msg_iter.type & ~(READ | WRITE), status->iter_type);
+
+        if ((msg.msg_iter.type & ~(READ | WRITE)) == status->iter_type) {
+            char *data = bpf_map_lookup_elem(&write_buffer_heap, &zero);
+            if (data != NULL) {
+                if(!is_v4_or_v6(sk, status)){
```

Comment:
I would move it to just after the flag check (`if (status->protocol_inspection_enabled) {`). 
Another thing, should we just `return 0`? Shouldn't we `return increment_tcp_stats(sk, status, size, 0);`?

Reply:
Oh, true! This whole block should not "return"

Reply:
There was already a `return 0;` in the `tcp_send()` function.  Should I replace with `return increment_tcp_stats(sk, status, size, 0);`?  If it is not ip4 or ip6, do we call `return increment_tcp_stats(sk, status, size, 0);` or `return 0;`?

Reply:
Maybe you could do on line 899:
```
    if (status->protocol_inspection_enabled && is_v4_or_v6(sk, status)) {
```

Reply:
@julianosk we checked what `increment_tcp_stats(sk, status, size, 0);` actually does.  If not ip4 or ip6, it does nothing.  So we move the check to above the `if (status->protocol_inspection_enabled)` check, and it returns 0.